datafusion/core/src/physical_planner.rs

```diff
@@ -496,776 +550,966 @@ impl DefaultPhysicalPlanner {
+            // Leaves (no children)
+            LogicalPlan::TableScan(TableScan {
+                source,
+                projection,
+                filters,
+                fetch,
+                ..
+            }) => {
+                let source = source_as_provider(source)?;
+                // Remove all qualifiers from the scan as the provider
+                // doesn't know (nor should care) how the relation was
+                // referred to in the query
+                let filters = unnormalize_cols(filters.iter().cloned());
+                source
+                    .scan(session_state, projection.as_ref(), &filters, *fetch)
+                    .await?
+            }
+            LogicalPlan::Values(Values { values, schema }) => {
+                let exec_schema = schema.as_ref().to_owned().into();
+                let exprs = values
+                    .iter()
+                    .map(|row| {
+                        row.iter()
+                            .map(|expr| {
+                                self.create_physical_expr(expr, schema, session_state)
                             })
                             .collect::<Result<Vec<Arc<dyn PhysicalExpr>>>>()
-                        })
-                        .collect::<Result<Vec<_>>>()?;
-                    let value_exec = ValuesExec::try_new(
-                        SchemaRef::new(exec_schema),
-                        exprs,
-                    )?;
-                    Ok(Arc::new(value_exec))
-                }
-                LogicalPlan::Window(Window {
-                    input, window_expr, ..
-                }) => {
-                    if window_expr.is_empty() {
-                        return internal_err!(
-                            "Impossibly got empty window expression"
-                        );
+                    })
+                    .collect::<Result<Vec<_>>>()?;
+                let value_exec = ValuesExec::try_new(SchemaRef::new(exec_schema), exprs)?;
+                Arc::new(value_exec)
+            }
+            LogicalPlan::EmptyRelation(EmptyRelation {
+                produce_one_row: false,
+                schema,
+            }) => Arc::new(EmptyExec::new(SchemaRef::new(
+                schema.as_ref().to_owned().into(),
+            ))),
+            LogicalPlan::EmptyRelation(EmptyRelation {
+                produce_one_row: true,
+                schema,
+            }) => Arc::new(PlaceholderRowExec::new(SchemaRef::new(
+                schema.as_ref().to_owned().into(),
+            ))),
+            LogicalPlan::DescribeTable(DescribeTable {
+                schema,
+                output_schema,
+            }) => {
+                let output_schema: Schema = output_schema.as_ref().into();
+                self.plan_describe(schema.clone(), Arc::new(output_schema))?
+            }
+
+            // 1 Child
+            LogicalPlan::Copy(CopyTo {
```

Comment:
As some follow on PR, we could refactor this logic out of one giant match statement into functions like

```rust
match plan { 
...
    LogicalPlan::Copy(copy_to) => copy_to_physical(copy_to),
...
}
```

But after this PR that refactoring seems like it would mostly improve readability rather than any stack usage